client/app/scripts/components/nodes.js

```diff
@@ -73,7 +73,15 @@ class Nodes extends React.Component {
   }
 
   setDimensions() {
-    this.props.setViewportDimensions(window.innerWidth, window.innerHeight);
+    // Use the `scope-app` div to determine the viewport rectangle, because `window` gives
+    // wrong values for scope container inside Weave Cloud (on dev & prod), because of the
+    // top navigation toolbar that is counted in, even though it's not part of Scope.
+    const viewport = document.getElementsByClassName('scope-app')[0].getBoundingClientRect();
+    // Not sure why we need to subtract `left` and `top`, but that gives us correct values.
```

Comment:
Removing that height didn't really help, but it's true that we should set the `height` of the `scope-app` correctly in Weave Cloud, rather than trying to approximate its position from within Scope UI.

Subtracting `left` and `top` like I did above happened to work in Weave Cloud, but wouldn't necessarily give good results when someone would randomly position Scope iFrame within their app.

So a proper solution is to `setViewportDimensions` directly from `width` and `height` here and to correct the `scope-app` height in the `service-ui` to account for the navigation toolbar, like this:

```
.scope-app {
  height: calc(100% - #{$top-toolbar-height});
}
```

I will add this piece of code to Service UI when I include these latest changes to the Scope UI (including this story) so that they are both pushed at once.